src/projection.jl

```diff
@@ -21,7 +21,19 @@ struct ProjectTo{P,D<:NamedTuple}
     info::D
 end
 ProjectTo{P}(info::D) where {P,D<:NamedTuple} = ProjectTo{P,D}(info)
-ProjectTo{P}(; kwargs...) where {P} = ProjectTo{P}(NamedTuple(kwargs))
+
+# We'd like to write
+# ProjectTo{P}(; kwargs...) where {P} = ProjectTo{P}(NamedTuple(kwargs))
+#
+# but the kwarg dispatcher has non-trivial complexity. See rules.jl for an
+# explanation of this trick.
+const empty_nt = NamedTuple()
+ProjectTo{P}() where {P} = ProjectTo{P}(empty_nt)
+
+const Type_kwfunc = Core.kwftype(Type).instance
```

Comment:
```suggestion
# Performance hack to make kwargs faster
const Type_kwfunc = Core.kwftype(Type).instance
```

Reply:
There's a comment above that explains this.